src/modules/NtpClient.rb

```diff
@@ -1263,56 +998,294 @@ def AutoPackages
+          # progress step
+          _("Finished")
+        ],
+        ""
+      )
+    end
+
+    def write_ntp_conf
+      conf = SCR.Read(path(".etc.ntp_conf.all"))
+
+      return false if conf.nil?
+
+      conf["value"] = records_for_write(@ntp_records)
+
+      return false if !SCR.Write(path(".etc.ntp_conf.all"), conf)
+
+      ret = SCR.Write(path(".etc.ntp_conf"), nil)
+
+      FileChanges.StoreFileCheckSum("/etc/ntp.conf")
+
+      ret
+    end
+
+    def write_ntp_policy
+      # write policy and run netconfig command
+      SCR.Write(
+        path(".sysconfig.network.config.NETCONFIG_NTP_POLICY"),
+        @ntp_policy
+      )
+      SCR.Write(path(".sysconfig.network.config"), nil)
+    end
+
+    def write_chroot_config
+      SCR.Write(
+        path(".sysconfig.ntp.NTPD_RUN_CHROOTED"),
+        @run_chroot ? "yes" : "no"
+      )
+
+      SCR.Write(path(".sysconfig.ntp"), nil)
+    end
+
+    def update_netconfig
+      error = SCR.Execute(path(".target.bash"), "/sbin/netconfig update -m ntp") != 0
+
+      Report.Error(_("Cannot update the dynamic configuration policy.")) if error
+
+      !error
+    end
+
+    def check_service
+      adjusted = @run_service ? Service.Enable(@service_name) : Service.Disable(@service_name)
+
+      # error report
+      Report.Error(Message.CannotAdjustService("NTP")) unless adjusted
+
+      if @run_service
+        unless @write_only
+          # error report
+          Report.Error(_("Cannot restart the NTP daemon.")) unless Service.Restart(@service_name)
+        end
+      else
+        Service.Stop(@service_name)
+      end
+    end
+
+    def update_cron_settings!
+      if @synchronize_time
+        SCR.Write(
+          path(".target.string"),
+          @cron_file,
+          "-*/#{@sync_interval} * * * * root /usr/sbin/start-ntpd ntptimeset &>/dev/null\n"
+        )
+      else
+        SCR.Execute(
+          path(".target.bash"),
+          "test -e #{@cron_file} && rm #{@cron_file};"
+        )
+      end
+    end
+
+    def record_to_h(record)
+      {
+        "comment" => record["comment"],
+        "kind"    => "value",
+        "name"    => record["type"] == "__clock" ? "server" : record["type"],
+        "type"    => 0,
+        "value"   => "#{record["address"]} #{record["options"]}"
+      }
+    end
+
+    def fugde_options_to_h(record)
+      {
+        "comment" => record["fudge_comment"],
+        "kind"    => "value",
+        "name"    => "fudge",
+        "type"    => 0,
+        "value"   => "#{record["address"]} #{record["fudge_options"]}"
+      }
+    end
+
+    def append_restrict_records!(records)
+      @restrict_map.each do |key, m|
+        options = " "
+        options << "mask #{m["mask"]} " if !m["mask"].to_s.empty?
+        options << m["options"].to_s
+        ret = {
+          "address" => key,
+          "comment" => m["comment"].to_s,
+          "type"    => "restrict",
+          "options" => options
+        }
+        records << ret
```

Comment:
You're modifying an argument. I would prefer one of these options:

1) Don't modify the argument: `@ntp_records += restrict_records`
2) Modify `@ntp_records` directly (after all, it's a instance variable).